sql/core/src/test/resources/sql-tests/results/literals.sql.out

```diff
@@ -0,0 +1,136 @@
+-- Automatically generated by SQLQueryTestSuite
+-- Number of queries: 16
+
+
+-- !query 0
+select null
+-- !query 0 schema
+struct<NULL:null>
+-- !query 0 output
+NULL
+
+
+-- !query 1
+select true, false
+-- !query 1 schema
+struct<true:boolean,false:boolean>
+-- !query 1 output
+true	false
+
+
+-- !query 2
+select 1Y, 1 Y, 127 Y, -127 Y
+-- !query 2 schema
+struct<1:tinyint,Y:int,Y:int,Y:int>
+-- !query 2 output
+1	1	127	-127
+
+
+-- !query 3
+select 128Y
+-- !query 3 schema
+struct<>
+-- !query 3 output
+org.apache.spark.sql.catalyst.parser.ParseException
+
+Value out of range. Value:"128" Radix:10(line 1, pos 7)
+
+== SQL ==
+select 128Y
+-------^^^
+
+
+-- !query 4
+select 1S, 1 S, 32767 S, -32767 S
+-- !query 4 schema
+struct<1:smallint,S:int,S:int,S:int>
+-- !query 4 output
+1	1	32767	-32767
+
+
+-- !query 5
+select 32768 S
+-- !query 5 schema
+struct<S:int>
+-- !query 5 output
+32768
+
+
+-- !query 6
+select 1L, 1 L, 2147483648 L, 9223372036854775807 L
+-- !query 6 schema
+struct<1:bigint,L:int,L:bigint,L:bigint>
+-- !query 6 output
+1	1	2147483648	9223372036854775807
+
+
+-- !query 7
+select 9223372036854775808 L
+-- !query 7 schema
+struct<L:decimal(19,0)>
+-- !query 7 output
+9223372036854775808
+
+
+-- !query 8
+select 1, -1
+-- !query 8 schema
+struct<1:int,(-1):int>
+-- !query 8 output
+1	-1
+
+
+-- !query 9
+select 2147483647, -2147483648
+-- !query 9 schema
+struct<2147483647:int,(-2147483648):bigint>
+-- !query 9 output
+2147483647	-2147483648
+
+
+-- !query 10
+select 2147483648, -2147483649
+-- !query 10 schema
+struct<2147483648:bigint,(-2147483649):bigint>
+-- !query 10 output
+2147483648	-2147483649
+
+
+-- !query 11
+select 9223372036854775807, -9223372036854775808
+-- !query 11 schema
+struct<9223372036854775807:bigint,(-9223372036854775808):decimal(19,0)>
+-- !query 11 output
+9223372036854775807	-9223372036854775808
+
+
+-- !query 12
+select 9223372036854775808, -9223372036854775809
+-- !query 12 schema
+struct<9223372036854775808:decimal(19,0),(-9223372036854775809):decimal(19,0)>
+-- !query 12 output
+9223372036854775808	-9223372036854775809
+
+
+-- !query 13
+select 1D, 1 D, 1.2D, 1e10, 1.5e5, .10 D, 0.10 D, .1e5
```

Comment:
there is a bug here too.

I would expect .10 D to be parsed as double, not decimal.